include/zephyr/drivers/auxdisplay.h

```diff
@@ -0,0 +1,824 @@
+ */
+typedef int (*auxdisplay_backlight_set_api)(const struct device *dev,
+					    uint8_t backlight);
+
+/**
+ * @typedef	auxdisplay_is_busy_api
+ * @brief	Callback API to check if the display is busy with an operation
+ * See auxdisplay_is_busy() for argument description
+ */
+typedef int (*auxdisplay_is_busy_api)(const struct device *dev);
+
+/**
+ * @typedef	auxdisplay_custom_character_set_api
+ * @brief	Callback API to set a customer character on the display for usage
+ * See auxdisplay_custom_character_set() for argument description
+ */
+typedef int (*auxdisplay_custom_character_set_api)(const struct device *dev,
+						   struct auxdisplay_character *character);
+
+/**
+ * @typedef	auxdisplay_write_api
+ * @brief	Callback API to write text to the display
+ * See auxdisplay_write() for argument description
+ */
+typedef int (*auxdisplay_write_api)(const struct device *dev,
+				    const uint8_t *data, uint16_t len);
+
+/**
+ * @typedef	auxdisplay_custom_command_api
+ * @brief	Callback API to send a custom command to the display
+ * See auxdisplay_custom_command() for argument description
+ */
+typedef int (*auxdisplay_custom_command_api)(const struct device *dev,
+					     struct auxdisplay_custom_data *command);
+
+/**
+ * @brief Text display driver API
+ * API which a textual display driver should expose
+ */
+struct auxdisplay_driver_api {
+	auxdisplay_cursor_set_enabled_api cursor_set_enabled;
+	auxdisplay_position_blinking_set_enabled_api position_blinking_set_enabled;
+	auxdisplay_cursor_shift_set_api cursor_shift_set;
+	auxdisplay_cursor_position_set_api cursor_position_set;
+	auxdisplay_cursor_position_get_api cursor_position_get;
+	auxdisplay_display_position_set_api display_position_set;
+	auxdisplay_display_position_get_api display_position_get;
+	auxdisplay_capabilities_get_api capabilities_get;
+	auxdisplay_clear_api clear;
+	auxdisplay_brightness_get_api brightness_get;
+	auxdisplay_brightness_set_api brightness_set;
+	auxdisplay_backlight_get_api backlight_get;
+	auxdisplay_backlight_set_api backlight_set;
+	auxdisplay_is_busy_api is_busy;
+	auxdisplay_custom_character_set_api custom_character_set;
+	auxdisplay_write_api write;
+	auxdisplay_custom_command_api custom_command;
+};
+
+/**
+ * @brief		Set cursor enabled status on a text display
+ *
+ * @param dev		Pointer to device structure
+ * @param enabled	True to enable cursor, false to disable
+ *
+ * @retval		0 on success else negative errno code.
+ */
+__syscall int auxdisplay_cursor_set_enabled(const struct device *dev,
+					    bool enabled);
+
+static inline int z_impl_auxdisplay_cursor_set_enabled(const struct device *dev,
+						       bool enabled)
+{
+	struct auxdisplay_driver_api *api = (struct auxdisplay_driver_api *)dev->api;
+
+	if (!api->cursor_set_enabled) {
+		return -ENOSYS;
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	pm_device_state_get(dev, &state);
+
+	if (state == PM_DEVICE_STATE_SUSPENDED) {
+		return -EIO;
+	}
+#endif
```

Comment:
-1 on this, other options:

1. Devices can implement runtime PM
2. App keeps track of device state (fragile)

Reply:
Devices do implement PM but I thought it would have been better to put common parts here to check if it was in power down mode because if a display is powered down, you can't change its configuration (and if you can it's not really powered down)